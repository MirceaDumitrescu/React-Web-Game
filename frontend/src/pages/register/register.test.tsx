import RegisterComponent from './register'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import createStoreComponent from '../../features/utils/helpers'
import configs from './config_register'

const Register = createStoreComponent(RegisterComponent)

it('should match snapshot', () => {
  const { asFragment } = render(<Register />)
  expect(asFragment()).toMatchSnapshot()
})

it('should have all inputs from config', () => {
  render(<Register />)
  configs.forEach((config) => {
    expect(screen.getByTestId(config.name)).toBeInTheDocument()
  })
})
