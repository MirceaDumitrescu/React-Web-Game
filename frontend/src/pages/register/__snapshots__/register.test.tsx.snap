// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form__register"
  >
    <h4
      class="form__title"
    >
      Register Form
    </h4>
    <div
      class="form__group"
    >
      <form
        class="form"
      >
        <div
          class="form__group"
        >
          <label>
            Name
          </label>
          <input
            class="form__group__input"
            data-testid="username"
            name="username"
            type="text"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label>
            Email adress
          </label>
          <input
            class="form__group__input"
            data-testid="email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label>
            Password
          </label>
          <input
            class="form__group__input"
            data-testid="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label>
            Confirm Password
          </label>
          <input
            class="form__group__input"
            data-testid="confirmPass"
            name="confirmPass"
            type="password"
            value=""
          />
        </div>
        <div
          class="form__group"
        >
          <label />
          <input
            class="form__btn"
            data-testid="button"
            name="button"
            type="submit"
            value="Register"
          />
        </div>
      </form>
      <div
        class="form__group"
      >
        <button
          class="form__redirect"
          data-testid="form-button"
        >
          <a
            class="form__redirect__link"
            href="/login"
          >
            Already have an account? Login here!
          </a>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
